File: ProjectRSC/Development/Server/src/org/darkquest/gs/plugins/misc/Dummy.java
package org.darkquest.gs.plugins.misc;

import java.util.Arrays;

import org.darkquest.gs.event.MiniEvent;
import org.darkquest.gs.model.GameObject;
import org.darkquest.gs.model.Player;
import org.darkquest.gs.plugins.listeners.action.ObjectActionListener;
import org.darkquest.gs.plugins.listeners.executive.ObjectActionExecutiveListener;
import org.darkquest.gs.world.World;

public class Dummy implements ObjectActionExecutiveListener, ObjectActionListener {

	static int[] ids;
	static {
		ids = new int[] { 49 };
		Arrays.sort(ids);
	}
	
	@Override
	public boolean blockObjectAction(GameObject obj, String command, Player player) {
		return Arrays.binarySearch(ids, obj.getID()) >= 0;
	}

	@Override
	public void onObjectAction(GameObject object, String command, Player owner) {	
		if(Arrays.binarySearch(ids, object.getID()) >= 0) {
			owner.setBusy(true);
			owner.getActionSender().sendMessage("You swing at the dummy");

			World.getWorld().getDelayedEventHandler().add(new MiniEvent(owner, 3500) {
				public void action() {
					owner.setBusy(false);
					if (owner.getCurStat(0) > 7) {
						owner.getActionSender().sendMessage("There is only so much you can learn from hitting a dummy");
						return;
					}
					owner.getActionSender().sendMessage("You hit the dummy");
					owner.incExp(0, 5, true);
					owner.getActionSender().sendStat(0);
				}
			});
			return;
		}
	}

}

File: ProjectRSC/Server/scripts/python/misc/Dummy.py
from com.prsc.gs.plugins.listeners.action import ObjectActionListener
from com.prsc.gs.plugins.listeners.executive import ObjectActionExecutiveListener
from com.prsc.gs.plugins import PlugInterface

'''
@author: GORF
Called when a user is hitting a dummy in varrock
'''

class Dummy(PlugInterface, ObjectActionListener, ObjectActionExecutiveListener):
	
	DUMMY = 49
	MAX_LEVEL = 7
	
	def onObjectAction(self, gameObject, command, player):
		script = player.getScriptHelper()
		script.occupy()
		script.displayMessage("You swing at the dummy")	
		script.sleep(3500)
		
		if script.getCurrentLevel(player.SkillType.ATTACK) > self.MAX_LEVEL:
			script.displayMessage("There is only so much you can learn from hitting a dummy")
			script.release()
			return
		
		script.displayMessage("You hit the dummy")
		script.advanceStat(player.SkillType.ATTACK, 5, True)
		script.release()
	
	def blockObjectAction(self, gameObject, command, player):
		return gameObject.getID() == self.DUMMY and command == "hit"
